cpp/src/arrow/acero/asof_join_node.cc

```diff
@@ -786,10 +840,15 @@ class InputState : public util::SerialSequencingQueue::Processor {
   std::vector<col_index_t> key_col_index_;
   // Type id of the time column
   Type::type time_type_id_;
+  // Type id of the time column's index type or Type::NA if the time column
+  // is not a dictionary
+  Type::type time_index_type_id_;
```

Comment:
```suggestion
  Type::type time_dictionary_index_type_id_;
```
Nice docstring; I would encode more of the information in the variable name too so the code is a little easier to read.